modal/_utils/function_utils.py

```diff
@@ -234,7 +247,8 @@ def get_tag(self):
         return self.function_name
 
     def is_nullary(self):
-        for param in self._signature.parameters.values():
+        signature = inspect.signature(self.raw_f)
```

Comment:
Yes, but it would be nice if it was guaranteed by the type system somehow and not just by a runtime check... Might make sense to split out FunctionInfo and ClassServiceInfo eventually, although they are overlapping to a large degree